iceoryx_hoofs/test/moduletests/test_posix_thread.cpp

```diff
@@ -35,66 +38,113 @@ class Thread_test : public Test
+                             })
+                     .has_error());
+
+    start = std::chrono::steady_clock::now();
+    threadSync.notify();
+    sut.reset();
+    end = std::chrono::steady_clock::now();
+
+    EXPECT_THAT(std::chrono::nanoseconds(end - start).count(), Gt(TEST_WAIT_TIME.toNanoseconds()));
+}
+
 TEST_F(Thread_test, SetAndGetWithEmptyThreadNameIsWorking)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "b805a0a6-29c0-41df-b5b7-3f66499d151a");
+    ::testing::Test::RecordProperty("TEST_ID", "ba2ed4d9-f051-4ad1-a2df-6741134c494f");
+    ASSERT_FALSE(
+        ThreadBuilder().create(sut, [] { std::this_thread::sleep_for(std::chrono::milliseconds(10)); }).has_error());
     ThreadName_t emptyString = "";
 
-    setThreadName(m_thread->native_handle(), emptyString);
-    auto getResult = getThreadName(m_thread->native_handle());
+    sut->setName(emptyString);
+    auto getResult = sut->getName();
 
     EXPECT_THAT(getResult, StrEq(emptyString));
 }
 
 TEST_F(Thread_test, SetAndGetWithThreadNameCapacityIsWorking)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "115cf4e9-4c7a-4fcc-8df8-65e3b3b547d1");
+    ::testing::Test::RecordProperty("TEST_ID", "a67128fe-a779-4bdb-a849-3bcbfed4b20f");
+    ASSERT_FALSE(
+        ThreadBuilder().create(sut, [] { std::this_thread::sleep_for(std::chrono::milliseconds(10)); }).has_error());
     ThreadName_t stringEqualToThreadNameCapacitiy = "123456789ABCDEF";
     EXPECT_THAT(stringEqualToThreadNameCapacitiy.capacity(), Eq(stringEqualToThreadNameCapacitiy.size()));
 
-    setThreadName(m_thread->native_handle(), stringEqualToThreadNameCapacitiy);
-    auto getResult = getThreadName(m_thread->native_handle());
+    sut->setName(stringEqualToThreadNameCapacitiy);
+    auto getResult = sut->getName();
 
     EXPECT_THAT(getResult, StrEq(stringEqualToThreadNameCapacitiy));
 }
 
 TEST_F(Thread_test, SetAndGetSmallStringIsWorking)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "d6c2d0b5-a6ee-43e6-8870-053feb6de845");
+    ::testing::Test::RecordProperty("TEST_ID", "b5141d3c-2721-478c-b3d1-f35fb3321117");
+    ASSERT_FALSE(
+        ThreadBuilder().create(sut, [] { std::this_thread::sleep_for(std::chrono::milliseconds(10)); }).has_error());
     char stringShorterThanThreadNameCapacitiy[] = "I'm short";
 
-    setThreadName(m_thread->native_handle(), stringShorterThanThreadNameCapacitiy);
-    auto getResult = getThreadName(m_thread->native_handle());
+    sut->setName(stringShorterThanThreadNameCapacitiy);
+    auto getResult = sut->getName();
 
     EXPECT_THAT(getResult, StrEq(stringShorterThanThreadNameCapacitiy));
 }
+// uint32_t someFunction(uint32_t a)
+//{
+// return a;
+//}
+// TEST_F(Thread_test, Blub)
+//{
+// iox::cxx::function<void()> callable = [&]() { someFunction(3); };
+// iox::cxx::function<void()> callable2;
+// optional<Thread> myThread;
+// ThreadBuilder().create(myThread, callable2).expect("Couldn't create a thread.");
+//}
```

Comment:
Do you want to keep this?

Reply:
Whoops... I removed these lines since I had to rebase anyway.